checkov/terraform/graph_builder/foreach_handler.py

```diff
@@ -131,17 +136,97 @@ def _render_sub_graph(sub_graph: l_graph.TerraformLocalGraph, blocks_to_render:
+                        attrs[k][0] = attrs[k][0].replace("${" + key_to_change + "}", str(val_to_change))
+                        attrs[k][0] = attrs[k][0].replace(key_to_change, str(val_to_change))
+                elif isinstance(v, list) and len(v) == 1 and isinstance(v[0], list):
+                    for i, item in enumerate(v):
+                        if isinstance(item, str) and (key_to_change in item or "${" + key_to_change + "}" in item):
+                            if v[i] == "${" + key_to_change + "}":
+                                v[i] = val_to_change
+                            else:
+                                v[i] = item.replace("${" + key_to_change + "}", str(val_to_change))
+                                v[i] = v[i].replace(key_to_change, str(val_to_change))
+                elif isinstance(v, dict):
+                    self._update_attributes(v, {key_to_change: val_to_change})
+
+    @staticmethod
+    def _build_key_to_val_changes(new_val: str, new_key: str = ''):
+        key_to_val_changes: dict[str, str] = {
+            EACH_VALUE: new_val,
+            EACH_KEY: new_val
+        }
+        if new_key:
+            key_to_val_changes[EACH_KEY] = new_key
+        return key_to_val_changes
+
+    def _create_new_resource(self, main_resource: TerraformBlock, new_value: int | str, new_key: str = ''):
+        new_resource = deepcopy(main_resource)
+        block_type, block_name = new_resource.name.split('.')
+        if main_resource.attributes.get(COUNT_STRING):
+            self._update_attributes(new_resource.attributes, {COUNT_KEY: new_value})
+            self._update_attributes(new_resource.config.get(block_type, {}).get(block_name, {}), {COUNT_KEY: new_value})
+        elif main_resource.attributes.get(FOREACH_STRING):
+            key_to_val_changes = self._build_key_to_val_changes(new_value, new_key)
+            self._update_attributes(new_resource.attributes, key_to_val_changes)
+            self._update_attributes(new_resource.config.get(block_type, {}).get(block_name, {}), key_to_val_changes)
+        idx_to_change = new_key or new_value
+        self._add_index_to_block_properties(new_resource, idx_to_change)
+        self.local_graph.vertices.append(new_resource)
+
+    def _create_new_resources_foreach(self, statement: list[str] | dict[str, Any], main_resource: TerraformBlock) -> None:
+        if isinstance(statement, list):
+            for new_value in statement:
+                self._create_new_resource(main_resource, new_value, new_key='')
+        if isinstance(statement, dict):
+            for new_key, new_value in statement.items():
+                self._create_new_resource(main_resource, new_value, new_key=new_key)
+
+    def _delete_main_resource(self, block_index_to_statement: FOR_EACH_BLOCK_TYPE) -> None:
+        new_vertices = [
+            block for i, block in enumerate(self.local_graph.vertices) if i not in block_index_to_statement or not block_index_to_statement[i]
```

Comment:
in which case the block_index_to_statement[i] will be None?

Reply:
If we didn’t manage to get the foreach/count statement

Reply:
so why is it still on the list in this case?

Reply:
the func gets all the None/should render values, but maybe we failed to render some of them